dags/bluetooth_check_readers.py

```diff
@@ -0,0 +1,127 @@
+from airflow import DAG
+from datetime import datetime, timedelta
+from airflow.operators.python_operator import PythonOperator
+from airflow.operators.postgres_operator import PostgresOperator
+from airflow.hooks.postgres_hook import PostgresHook
+from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+from psycopg2 import sql
+from psycopg2.extras import execute_values
+from psycopg2 import connect, Error
+import logging
+import configparser
+
+LOGGER = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
+
+#to connect to pgadmin bot
+bt_postgres = PostgresHook("bt_bot")
+
+con = bt_postgres.get_conn()
+broken_list = []
+# check if the blip data was aggregated into aggr_5min bins as of yesterday.
+# if the aggr_5min table has at least one record for yesterday, it means pipeline is working 
+def pipeline_check(con):
+    with con.cursor() as cursor:
+        select_query1 = '''SELECT MAX (datetime_bin)::date from bluetooth.aggr_5min'''
+        cursor.execute(select_query1)
+        latest_date = cursor.fetchone()
+        #please suggest if this try except works
+        try:
+            if (latest_date[0]) >= (datetime.now().date() - timedelta(1)):
+                pass
+            else:
+                raise Exception ('There is no data in bluetooth.aggr_5min for yesterday')
+        except:
+            pass
+
+def broken_readers(con, check_date):
+    with con.cursor() as cursor: 
+        select_query2 = '''SELECT * from bluetooth.broken_readers(%s::date)'''
+        cursor.execute(select_query2, (check_date,))
+        broken_readers = cursor.fetchall()
+        broken_list.append(broken_readers)
+        num_broken = len(broken_list)
+        #please suggest if this try except works
+        try:
+            if broken_readers[0][0] == '':
+                pass
+            else:
+                raise Exception ('Some readers are down. Please check broken_readers_log')
+        except:
+            pass    
+           
+        
+SLACK_CONN_ID = 'slack_data_pipeline'
+def task_fail_slack_alert(context):
+    slack_webhook_token = BaseHook.get_connection(SLACK_CONN_ID).password
+    
+    if context.get('task_instance').task_id == 'task1':
+        task_msg = """:among_us_dead: Task {task} in blip check update failed, 
```

Comment:
This should read "No bluetooth data was found in the database for this date. Remote desktop to the terminal server and check the `blip_api` log".